src/ingest_validation_tests/fastq_validator_logic.py

```diff
@@ -13,45 +13,71 @@
 import fastq_utils
 from typing_extensions import Self
 
-filename_pattern = namedtuple("filename_pattern", ["prefix", "read_type", "set_num"])
+filename_pattern = namedtuple("filename_pattern", ["before_read", "read", "after_read"])
 
 
 def is_valid_filename(filename: str) -> bool:
     return bool(fastq_utils.FASTQ_PATTERN.fullmatch(filename))
 
 
+def get_filename(pattern: filename_pattern) -> str:
+    return f"{pattern.before_read}{pattern.read}#{pattern.after_read}"
+
+
 def get_prefix_read_type_and_set(filename: str) -> Optional[filename_pattern]:
     """
-    Looking for fastq filenames matching pattern:
-    <prefix>_<lane:L#+>_<read_type:I,R,read>#_<set_num:#+>
-    e.g. arbitrary_string_L001_R1_001.fastq
+    Looking for fastq filenames with a particular format to compare record counts.
+
+    Expected pattern:
+        - <arbitrary_text>_<lane:L#+>_<read_type:I,R,read>#_<set_num:#+>.<fastq,fastq.gz,fq>
+        - e.g. arbitrary_string_L001_R1_001.fastq
+    Minimum required elements: lane (must occur before read), read
+    May also include: arbitrary text, set_num
 
     Regex documentation:
-        PREFIX
-            - (?P<prefix> | named capture group "prefix"
-            - .*(?:L\\d+) | match anything before and including pattern L# (where # represents 1 or more digits)
-            - only capture prefix if read_type and set_num found
-        READ_TYPE
-            - (?=_ | assert that the character "_" is present at the beginning of this next group
-            - (?:(?P<read_type>R|read(?=[123])|I(?=[12]))) | only capture above match if followed by the
-               sequence _[R1,R2,R3,read1,read2,read3,I1,I2]; capture R/I/read as read_type, discarding digit
-        SET_NUM
-            - (?:\\d_ | ensure presence of "#_" before group, ignore characters
-            - (?P<set_num>\\d+) | capture group set_num of 1 or more digits
-               Note: if set_num needs to be optional, could change this portion to (?:\\d_?(?P<set_num>\\d+)|)
+        BEFORE_READ (?P<before_read>.*_L\\d{3}_.*(...))
```

Comment:
It seems like the previous regex captured an arbitrary number of digits after L, where-as the new version only allows for 3.

Reply:
Done.